### PR TITLE
error reporting fix

### PR DESCRIPTION
**Pull request by matei-p**

### CODE REVIEW

File: arwen/contexts/runtime.go [hunk 1]
```diff
@@ -518,7 +518,12 @@ func (context *runtimeContext) FailExecution(err error) {
 
 	var message string
 	if err != nil {
-		message = err.Error()
+		wrappedError, ok := (err.(arwen.WrappableError))
```
Comment (iulianpascalau) [inline arwen/contexts/runtime.go:521]: can be simplified as:
```
wrappedError, ok := err.(arwen.WrappableError)
```

Comment (matei-p) [reply arwen/contexts/runtime.go:521]: err is not always a WrappableError if this is the assumption

**Review by sasurobert — APPROVED** [review]

File: arwen/contexts/runtime.go (hunk 1 above)
Comment (sasurobert) [reply arwen/contexts/runtime.go:521]: the paranthesis are duplicated.

Comment (iulianpascalau) [reply arwen/contexts/runtime.go:521]: no, please remove the redundant parenthesis around the type assertion. It raises linter issues.

**Review by iulianpascalau — APPROVED** [review]

File: arwen/contexts/runtime.go [hunk 2]
```diff
@@ -531,7 +531,7 @@ func (context *runtimeContext) FailExecution(err error) {
 	context.host.Output().SetReturnMessage(message)
 	context.SetRuntimeBreakpointValue(arwen.BreakpointExecutionFailed)
 
-	logRuntime.Trace("execution failed", "message", message)
+	logRuntime.Trace("execution failed", "message", err.Error())
```
Comment (iulianpascalau) [inline arwen/contexts/runtime.go:534]: this can panic

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: arwen/contexts/runtime.go (hunk 2 above)
Comment (sasurobert) [reply arwen/contexts/runtime.go:534]: better add an if before trace. this call is not really called with err = nil. but it is better to be safe

**Review by iulianpascalau — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**